src/Microsoft.Data/DataFrame.cs

```diff
@@ -326,6 +339,41 @@ private DataFrame Clone(BaseColumn mapIndices = null, bool invertMapIndices = fa
 
         public void SetColumnName(BaseColumn column, string newName) => _table.SetColumnName(column, newName);
 
+        /// <summary>
+        /// Generates a concise summary of each column
+        /// </summary>
+        public DataFrame Info()
+        {
+            DataFrame ret = new DataFrame();
+            if (ColumnCount == 0)
+                return ret;
+
+            for (int i = 0; i < ColumnCount; i++)
+            {
+                BaseColumn column = Column(i);
+                DataFrame columnInfo = column.Info();
+                if (i == 0)
+                    ret = columnInfo;
```

Comment:
Why create a `ret = new DataFrame()` above, just to discard it here?

Also, this feels like a lot of allocations (a new DataFrame for every column) and then they get thrown away. Why not pass the DataFrame into `column.Info()` and have the column add to the current DataFrame?

Reply:
Fixed